src/xdp-utils.c

```diff
@@ -801,8 +840,15 @@ parse_app_info_from_snap (pid_t pid, GError **error)
       return NULL;
     }
 
+  app_name = g_key_file_get_string (metadata, SNAP_METADATA_GROUP_INFO,
+                                    SNAP_METADATA_KEY_APP_NAME, error);
+  if (app_name == NULL)
+    {
+      return NULL;
+    }
+
   app_info = xdp_app_info_new (XDP_APP_INFO_KIND_SNAP);
-  app_info->id = g_strconcat ("snap.", snap_name, NULL);
+  app_info->id = g_strconcat ("snap.", snap_name, "_", app_name, NULL);
```

Comment:
I think changing the ID like this is a bad idea. The reason I only used the snap name in `app_info->id` was that there is effectively no isolation between different applications within the same snap package. If the user grants access to some information to one application within the snap, there is nothing to prevent that app from passing that information on to the second app. So it makes most sense to record the permissions at the level of the package, rather than application.

I know there are rough edges in the portal UI for snaps at present, but I think a lot of this is better handled by passing desktop file IDs to the impl services (since the impl service isn't generally making its own access decisions, and is usually expecting a desktop ID anyway).